examples/google-cloud-speech/output/google-cloud-speech/lib/google/cloud/speech/v1/speech_client.rb

```diff
@@ -0,0 +1,411 @@
+            )
+            operation.on_done { |operation| yield(operation) } if block_given?
+            operation
+          end
+
+          # Performs bidirectional streaming speech recognition: receive results while
+          # sending audio. This method is only available via the gRPC API (not REST).
+          #
+          # @param reqs [Enumerable<Google::Cloud::Speech::V1::StreamingRecognizeRequest>]
+          #   The input requests.
+          # @param options [Google::Gax::CallOptions]
+          #   Overrides the default settings for this call, e.g, timeout,
+          #   retries, etc.
+          # @return [Enumerable<Google::Cloud::Speech::V1::StreamingRecognizeResponse>]
+          #   An enumerable of Google::Cloud::Speech::V1::StreamingRecognizeResponse instances.
+          #
+          # @raise [Google::Gax::GaxError] if the RPC is aborted.
+          #
+          # @note
+          #   EXPERIMENTAL:
+          #     Streaming requests are still undergoing review.
+          #     This method interface might change in the future.
+          #
+          # @example
+          #   require "google/cloud/speech"
+          #
+          #   speech_client = Google::Cloud::Speech.new(version: :v1)
+          #   request = {}
+          #   requests = [request]
+          #   speech_client.streaming_recognize(requests).each do |element|
+          #     # Process element.
+          #   end
+
+          def streaming_recognize reqs, options: nil
+            request_protos = reqs.lazy.map do |req|
+              Google::Gax::to_proto(req, Google::Cloud::Speech::V1::StreamingRecognizeRequest)
+            end
+            @streaming_recognize.call(request_protos, options)
+          end
+
+          ##
+          # Gets the latest state of a long-running operation. Clients can use
+          # this method to poll the operation result at intervals as recommended
+          # by the API service.
+          #
+          # @param name [String]
+          #   The name of the operation resource.
+          # @param options [Google::Gax::CallOptions]
+          #   Overrides the default settings for this call, e.g, timeout,
+          #   retries, etc.
+          # @return [Google::Gax::Operation]
+          # @raise [Google::Gax::GaxError] if the RPC is aborted.
+          # @example
+          #   require "google/cloud/speech"
+          #
+          #   speech_client = Google::Cloud::Speech.new version: :v1
+          #
+          #   op = speech_client.get_operation "-"
+          #
+          #   # Process error operations.
+          #   log_error op.error if op.error?
+          #
+          #   if op.done?
+          #     # Process completed operations.
+          #     log_finished op.response, op.metadata
+          #   else
+          #     # Process pending operations.
+          #     log_pending op.name, op.metadata
+          #   end
+          #
+          def get_operation name, options: nil
```

Comment:
Do we want to consider adding any of the other operation methods here (i.e. cancel)?

Reply:
We discussed this previously. The only two methods we might add to the client were `get_operation` and `list_operations`. But Speech doesn't implement the list API, so we don't have it here.

The other API methods are accessible on the `Google::Gax::Operation` object, and not needed on the client.

Reply:
Oh, that's right. We don't have a reliable way to know if `list` is implemented though, so that's still an issue. I will see if that's something we can append to the configuration somehow.

Reply:
Fun fact, the Container API has `get_operation` and `list_operations` methods, but they are not the long running operation kind.

Reply:
I suspected that we'd run into this. I'm also wondering if we'll ever hit it with the resource name helper methods (one of the reasons I was alway less enthusiastic about them).

How do you think we should handle these cases. We can certainly detect when collisions happen if we have a strategy for dealing with it.

Reply:
I'm not sure it will be a problem, actually. But I don't know how the server is implemented. Are the long-running operation methods implemented the same way the IAM methods are? Are they simply added to the API? If so, then the base service would need to ensure the method names don't conflict.

Reply:
The fully qualified name of the rpc operation is different in the two cases so I don't think it creates a conflict, and the container API probably wouldn't work now if it did. I wouldn't be surprised if it was considered a bad naming practice though.